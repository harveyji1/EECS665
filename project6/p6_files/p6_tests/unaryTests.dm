func : () void{
    myB : bool = true;
    myA : int = 1;
    if(!(myB)){
        myA = -1;
    }
}