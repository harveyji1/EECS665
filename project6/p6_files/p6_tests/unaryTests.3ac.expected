[BEGIN GLOBALS]
func
[END GLOBALS]
[BEGIN func LOCALS]
myB (local var of 8 bytes)
myA (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
[END func LOCALS]
fun_func:   enter func
            [tmp0] := NOT64 [myB]
            IFZ [tmp0] GOTO lbl_1
            [tmp1] := NEG64 1
            [myA] := [tmp1]
lbl_1:      nop
lbl_0:      leave func

